Applications/Spire/Source/Ui/DropDownMenuItem2.cpp

```diff
@@ -0,0 +1,51 @@
+#include "Spire/Ui/DropDownMenuItem2.hpp"
+#include <QMouseEvent>
+#include "Spire/Spire/Dimensions.hpp"
+
+using namespace boost::signals2;
+using namespace Spire;
+
+DropDownMenuItem2::DropDownMenuItem2(QVariant value, QWidget* parent)
+    : QWidget(parent),
+      m_value(std::move(value)),
+      m_is_highlighted(false) {
+  setAttribute(Qt::WA_Hover);
+}
+
+void DropDownMenuItem2::enterEvent(QEvent* event) {
+  m_hovered_signal();
+}
+
+void DropDownMenuItem2::mousePressEvent(QMouseEvent* event) {
+  if(event->button() == Qt::LeftButton) {
```

Comment:
What about pressing ENTER? Doesn't that also select an item?

Reply:
Under the combo_box specs, for the dropdown_button flow, it shows that the space key and mouse button selects an item. I would have to consult with Darryl but I thought a change was made where the space key was used as the key equivalent of a mouse click, and the enter key was used to submit an input value (like the text input).

Reply:
Wait, wrong widget.

In this case, because the DropDownMenu controls the 'current' item the DropDownMenuItems themselves don't take focus.

Reply:
What is the difference between selecting an item and submitting an input value?

Reply:
Darryl clarified the different functions in the case notes because I had them confused. The test menu input has them correct, where both enter and space will select an item, but only space will show the menu.

Reply:
In terms of key inputs, there shouldn't be a difference between selection and submission for items and values.

Reply:
What part of the code handles key presses?

Reply:
The test widget handles the events on behalf of the test label.

Reply:
Pressing ENTER should select the current item, go ahead and add that functionality.

Reply:
It currently does. I had some details confused but in response to the original comment, the DropDownMenuItems don't take focus so the enter key event will never be sent to them.

Reply:
Where is that implemented? Why doesn't the ```DropDownMenu``` have focus?

Reply:
Okay, I see. I thought the menu inputs would handle all key press events so I left them out of the test input for brevity.

I'll add the key events to the DropDownMenu for when it's the active window.

Reply:
Okay, I also suspect ```DropDownMenuItem```s need to handle ENTER key presses as well.

Reply:
I don't think there's a case where they take focus. The user can click on them so they handle their own mouse press events, but the parent input and menu is what has focus/is controlling the current item, so the input and menu use the current index to select the current item on a key press selection.

Reply:
Why don't they take focus?

Reply:
The typical case is that the parent input has focus, so I don't think giving them focus would provide a benefit.

Reply:
Have ```DropDownMenuItem``` signal that it's selected on ENTER and on mouse click.